### PR TITLE
*: provide a storage interface

### PR DESCRIPTION
**Pull request by vbatts**
Based on the memstore sample, provide a golang interface for how to
interact with a storage implementation.

Signed-off-by: Vincent Batts <vbatts@hashbangbash.com>

### CODE REVIEW

File: server-go/storage.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
+package server
+
+import (
+	"github.com/grafeas/grafeas/samples/server/go-server/api"
+	"github.com/grafeas/grafeas/samples/server/go-server/api/server/errors"
+)
+
+// Storager implementations are
```
Comment (R2wenD2) [inline server-go/storage.go]: it looks like this comment didn't get finished?

**Comment by vbatts** [comment]
@R2wenD2 what are these merge commits added to my branch? can i just rebase on master?

File: server-go/storage.go (hunk 1 above)
Comment (vbatts) [reply server-go/storage.go]: updated

**Comment by vbatts** [comment]
updated. PTAL

**Comment by R2wenD2** [comment]
Yeah, go ahead and rebase. We renamed a file, I was trying to avoid it impacting you too much :)

**Comment by vbatts** [comment]
I can work on `errors` next. Ideally, a top-level won't import from `./samples` :-)

**Comment by R2wenD2** [comment]
That sounds good. Thanks for the cleanup :0) I'd like to rename the swagger package and hoist that out of samples too.

**Comment by vbatts** [comment]
I was going to look at that too, but need to get familiar with that swagger
codegen process.

**Pull request merged**